frontends/web/src/routes/buy/info.tsx

```diff
@@ -97,7 +84,10 @@ class BuyInfo extends Component<Props, State> {
     if (options === undefined) {
       return <Spinner text={t('loading')} />;
     }
-    const name = this.getCryptoName();
+
+    const account = getAccount(this.props.accounts, this.props.code);
+    const name = getCryptoName(this.props.t, account);
+
```

Comment:
I think it is better to make props.code mandatory in this component, it should always use a code or not?

Then code should be mandatory in getAccount.

Didn't test but TypeScript seems happy:

```diff
diff --git a/frontends/web/src/routes/account/utils.ts b/frontends/web/src/routes/account/utils.ts
index 584d48bf0..7fb42d74d 100644
--- a/frontends/web/src/routes/account/utils.ts
+++ b/frontends/web/src/routes/account/utils.ts
@@ -17,10 +17,7 @@
 
 import { CoinCode, ScriptType, IAccount } from '../../api/account';
 
-export function getAccount(accounts: IAccount[], accountCode: string | undefined): IAccount | undefined {
-  if (!accounts) {
-    return undefined;
-  }
+export function getAccount(accounts: IAccount[], accountCode: string): IAccount | undefined {
   return accounts.find(({ code }) => accountCode === code);
 }
 
diff --git a/frontends/web/src/routes/buy/info.tsx b/frontends/web/src/routes/buy/info.tsx
index efd3f6adb..5fe798f24 100644
--- a/frontends/web/src/routes/buy/info.tsx
+++ b/frontends/web/src/routes/buy/info.tsx
@@ -28,7 +28,7 @@ import style from './info.module.css';
 
 interface BuyInfoProps {
     accounts: IAccount[];
-    code?: string;
+    code: string;
 }
 
 interface Option {
diff --git a/frontends/web/src/routes/router.tsx b/frontends/web/src/routes/router.tsx
index 5850ae52d..42b904a5b 100644
--- a/frontends/web/src/routes/router.tsx
+++ b/frontends/web/src/routes/router.tsx
@@ -81,6 +81,7 @@ export const AppRouter = ({ devices, deviceIDs, devicesKey, accounts, activeAcco
 
   const BuyInfoEl = <InjectParams>
     <BuyInfo
+      code={''}
       accounts={activeAccounts} />
   </InjectParams>;
 ```

Reply:
Good point, done!